src/ipc/handler.c

```diff
@@ -720,16 +721,20 @@ static int ipc_dma_trace_config(uint32_t header)
 	uint32_t ring_size;
 #endif
 	struct sof_ipc_dma_trace_params_ext params;
+	uint64_t ticks;
 	int err;
 
 	/* copy message with ABI safe method */
 	IPC_COPY_CMD(params, _ipc->comp_data);
 
-	if (iCS(header) == SOF_IPC_TRACE_DMA_PARAMS_EXT)
-		platform_timer->delta = params.timestamp -
-					platform_timer_get(platform_timer);
-	else
-		platform_timer->delta = 0;
+	if (iCS(header) == SOF_IPC_TRACE_DMA_PARAMS_EXT) {
+		ticks = clock_ns_to_ticks(platform_ts->clk,
+					  params.timestamp_ns);
+		platform_timer->delta_ns = ticks -
+					   platform_timer_get(platform_timer);
```

Comment:
assign ticks to nanosecond?

Reply:
I've also think that _ns is bit misleading here

Reply:
@bardliao external host IPC uses nanosecs, internal uses local time domain (which will be different for each platform)

Reply:
@bardliao 
I think the _ns is already with `timestamp_ns`. No need for delta.

Reply:
Yes, thanks everyone. The _ns here is a mistake. I will fix it then